### ISSUE TITLE
✅ 📊 🐞 Bug Report: Nested Carousel affects the performance of the Editor - [ED-12171]

### ISSUE THREAD

**Issue by sinanisler** [elementor/elementor#23704]
### Prerequisites

- [ ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ ] The issue still exists against the latest stable version of Elementor.

### Description

elementor new carousel feature making page editor loading/rendering very slow

I am guessing this is related to the nesting. new nesting is not very well coded. needs optimization and more testing.

**PLEASE for the love of god make more testing**...

### Steps to reproduce

make new carousel add 10 slide make advance multi container slides.

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [ ] I can reproduce this bug consistently following the steps above.

### System Info

doesnt matter this happens in every combination every possible situation. it happens in all of my dev sites.

make some big tests and advance sliders you will see it easily. MAKE MORE TESTS PLEASE.

<details>
<summary>Click to reveal</summary>

````txt

					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Debian 11 v10.5.19
	PHP Version: 7.4.33
	PHP Memory Limit: 4000M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 2000M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.3.1
	Site URL: https://dev.stan2.geopard-stuttgart.de
	Home URL: https://dev.stan2.geopard-stuttgart.de
	WP Multisite: No
	Max Upload Size: 2 GB
	Memory limit: 40M
	Max Memory limit: 4000M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: Europe/Berlin
	Debug Mode: Inactive

== Theme ==
	Name: Geopard.digital
	Version: 1.0
	Author: Geopard.digital
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.15.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.15.1
		Author: Elementor.com

	motion.page
		Version: 2.0.0
		Author: HypeWolf OÜ


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active
	Improved Asset Loading: Active
	Improved CSS Loading: Active
	Inline Font Icons: Inactive
	Additional Custom Breakpoints: Inactive
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active
	Grid Container: Inactive
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Editor Top Bar: Inactive
	Landing Pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Global Style Guide: Active by default
	Page Transitions: Active by default
	Notes: Inactive
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Active
	Taxonomy Filter: Inactive by default


== Log ==
	
PHP: showing 1 of 1PHP: 2023-06-29 13:25:41 [notice X 7][../wp-content/plugins/elementor/core/common/modules/ajax/module.php::175] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 5 of 5JS: 2023-08-02 18:15:44 [error X 3][https://dev.stan2.geopard-stuttgart.de/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.14.1:2:5688] Cannot read properties of undefined (reading 'autoplay') 
JS: 2023-08-02 18:16:06 [error X 5][https://dev.stan2.geopard-stuttgart.de/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.14.1:2:5777] Cannot read properties of undefined (reading 'params') 
JS: 2023-08-07 08:49:58 [error X 1][https://dev.stan2.geopard-stuttgart.de/wp-content/plugins/elementor/assets/js/web-cli.min.js?ver=3.15.1:3:107466] Routes: `panel/editor/actions` not found. 
JS: 2023-08-16 12:52:21 [error X 2][https://dev.stan2.geopard-stuttgart.de/wp-content/plugins/elementor/assets/js/7fada1a61d72da6d46ab.bundle.min.js:2:7826] Cannot read properties of null (reading 'dataset') 
JS: 2023-09-06 08:26:26 [error X 1][https://dev.stan2.geopard-stuttgart.de/wp-content/plugins/elementor-pro/assets/js/frontend.min.js?ver=3.15.1:2:11958] Failed to execute 'observe' on 'IntersectionObserver': parameter 1 is not of type 'Element'. 

Log: showing 20 of 252023-08-03 17:25:15 [info] elementor::elementor_updater Started 
2023-08-03 17:25:15 [info] Elementor/Upgrades - _on_each_version Start  
2023-08-03 17:25:15 [info] Elementor/Upgrades - _on_each_version Finished 
2023-08-03 17:25:15 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.14.1',
  'to' => '3.15.1',
)]
2023-08-03 17:25:15 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.14.1',
  'to' => '3.15.1',
)]
2023-08-11 09:37:23 [info] elementor::elementor_updater Started 
2023-08-11 09:37:23 [info] Elementor/Upgrades - _on_each_version Start  
2023-08-11 09:37:23 [info] Elementor/Upgrades - _on_each_version Finished 
2023-08-11 09:37:23 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.1',
  'to' => '3.15.2',
)]
2023-08-11 09:37:23 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.1',
  'to' => '3.15.2',
)]
2023-08-11 09:38:02 [info] elementor-pro::elementor_pro_updater Started 
2023-08-11 09:38:02 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-08-11 09:38:02 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-08-11 09:38:02 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.15.0',
  'to' => '3.15.1',
)]
2023-08-11 09:38:02 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.15.0',
  'to' => '3.15.1',
)]
2023-08-29 10:07:02 [info] elementor::elementor_updater Started 
2023-08-29 10:07:02 [info] Elementor/Upgrades - _on_each_version Start  
2023-08-29 10:07:02 [info] Elementor/Upgrades - _on_each_version Finished 
2023-08-29 10:07:02 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.2',
  'to' => '3.15.3',
)]
2023-08-29 10:07:02 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.2',
  'to' => '3.15.3',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	

				

````

</details>

**Comment by sinanisler** [comment]
ok pasted the system info

sadly I work only on client projects all of them in dev stage and under nda cant show more.
replicating the issue is not that hard create carousel ad lots of nested containers and images headings EASY. 
editor will start to lag carousel will start to lag and late render...


thanks nicholaszein

**Comment by nicholaszein** [comment]
Thank you for providing the missing information @sinanisler! 🙏 

Please allow us some time to investigate it, and get back to you with our findings. 🧐 

We appreciate your involvement in our community. For efficient troubleshooting, please remember to include the **<br /><ins>FULL, and essential</ins> [System Info](https://elementor.com/help/hare-my-system-info/)** with each issue reported. We'll need it to help you effectively. 

#### ⚠️ As a gentle reminder, issues without it may need to be closed without troubleshooting. Providing this information upfront aligns with our guidelines, saves time, and ensures swift resolution. Thank you for understanding.

Best regards

**Comment by portrayaloflife** [comment]
+1, also seeing page load impact with carousel added to a page

**Comment by sinanisler** [comment]
you gotta see multiple carousel usage in same page takes ages to render in editor 🤣🤣🤣

**Comment by MikeLikesThat** [comment]
> this nesting widgets rushed too much and it shows. sadly...

This is a Beta feature. There is always a chance for issues to pop up when updating. I would recommend to not implement this feature into live sites until the status has been upgraded.

**Comment by sinanisler** [comment]
freezing problems happening all loop grid or nesting new features not only carousel :)
but I get your point yes. agree.

**Comment by sinanisler** [comment]
just recently made a page with multiple nested accordions holy shit it is almost not editable.

my agency teams cant edit the page. rendering so laggy it is like there is infinite loop going on when editor tried to render this accordion nested containers :)

**all of these nested container features and editor render problems are related.**

**Comment by metal696heart** [comment]
This bug does [this](https://freeimage.host/i/JdDAzLF) . Nested indeed.

**Comment by Xavier1610** [comment]
Yes same problem here !

**Comment by doublebondweb** [comment]
This issue persists. 

Replicating this is extremely easy. A fresh WP installation with just Elementor and Elementor Pro, a Carousel widget (Nested Elements feature must be active) with multiple slides (8+) where each slide has a heading, text and button (for example). 

Doing anything to this Carousel is extremely hard. Be it changing the number of slides displayed, adding a new slide, or editing an existing slide... It just slows everything down to the point that it's just not possible to work.

**Comment by pan-athen** [comment]
I can also confirm that the new Carousel element slows down the Elementor Editor a lot. Even with 4-6 slides, it becomes impossible to work with.

I use it on three websites, and it behaves the same way every time. I lose about 15-20 minutes just to change the slide title on the properties and change some texts.

Publishing the page is slowed down, too, after making changes to the Carousel's content.

**Comment by rblackfox** [comment]
Can also confirm this. 8gb memory over allocated to php and WP, all plugins disabled it still happens, this makes working in the editor impossible.

**Comment by Temitayode** [comment]
This is still happening. Can they just fix this. So difficult editing when the loop carousel is there

**Comment by kemoboy** [comment]
Yep, this still happens. :( I have a design from a client where I have lots of nested items inside a carousel... editor stops loading at all sometimes... It's just a pain to work with. 

But hey, we have useless AI that generate some containers... why fix bugs that are more important? -_-

**Comment by pan-athen** [comment]
> But hey, we have useless AI that generate some containers... why fix bugs that are more important? -_-

I agree, and I'm sure most people would agree, the core functionality of the features should be at high quality before serving what is fashionable. Right now AI is bringing a lot of memberships and that's what investors wanna see, but as with any large platform, if Elementor doesn't focus on its core, a disruptive product will take their clients in a clean sweep.

I'm very happy with Elementor, but the feature should not be released at all if it works like that. They should know better.

**Comment by AleheyGo** [comment]
The problem is extremely urgent. After moving a 6-slide carousel from the staging page for development to the main page, the editor freezes so much that even after 30 minutes I can't publish the changes. It is also not possible to edit the page, since after any changes the entire page is recalculated. Need a fix for the problem!

**Comment by PhartBox** [comment]
Same problem here as @AleheyGo wrote...

**Comment by EstebanToloza** [comment]
This problem persists and severely affects the user experience when working with elementor.

**Comment by nicholaszein** [comment]
Hi, Nick from Elementor here.

For the past several months we've been gathering data and working non-stop on a huge update for all nested elements that should improve performance significantly. We expect to finish it soon but not with the release of 3.22.0 yet.

We understand your frustration, and thank you for your patience! 🙏

**Comment by kemoboy** [comment]
Maybe by the end of the year or next year, as we are used with bugs being fixed so late that we've already got accustomed to them. 

Maybe you should try instead of pushing the AI thingy and other things so much, to solve some actual problems that affects people in real world. :(

**Comment by lavendaniel** [comment]
This is still a problem. Can it be prioritised please?

**Comment by lucianodigital2b** [comment]
Guys, are nested carousel a thing or na? i have clients reporting that it was working but now they cant find the widget anymore.

**Comment by yeshourunr** [comment]
This is a huge problem still. 3.22.0 has already come out and the issue has only gotten worse. I've stopped recommending Elementor to clients — I can't do so in good faith when basic features are not functional. 

Editing a page with 1 loop/nested carousel and 8 slides takes 30 minutes and that's only adding 1 more slide. Double that when you add a period to a heading. Even basic editing features don't work anymore!

And let's not talk about nested accordions. Omg. Does Elementor only care about their hosting now? The plugin has no more QA and broken update after broken update is released with more AI features that nobody wants.

FIX YOUR BROKEN PLUGIN.

**Comment by maaxjahz** [comment]
> This is a huge problem still. 3.22.0 has already come out and the issue has only gotten worse. I've stopped recommending Elementor to clients — I can't do so in good faith when basic features are not functional.
> 
> Editing a page with 1 loop/nested carousel and 8 slides takes 30 minutes and that's only adding 1 more slide. Double that when you add a period to a heading. Even basic editing features don't work anymore!
> 
> And let's not talk about nested accordions. Omg. Does Elementor only care about their hosting now? The plugin has no more QA and broken update after broken update is released with more AI features that nobody wants.
> 
> FIX YOUR BROKEN PLUGIN.

Agreed, I have a website with a carousel with 28 seperate templates in it, website takes ages to load, it stays fully white untill the site is fully loaded but it just takes ages... Can't they just add a lazy load on the templates in the carousel and have only one load at a time?

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved with the release of Elementor v3.23.0! 🥳 

The time to interaction should improve by 60%-98%, and performance overall should be better. For a full breakdown please read [this article](https://developers.elementor.com/elementor-3-23-developers-update/#:~:text=Performant%20Nested%20Elements%20In%20the%20Editor).

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
